### PR TITLE
Show progress bar at first iteration

### PR DESCRIPTION
**Pull request by okuta**
```
epoch       main/loss   validation/main/loss  main/accuracy  validation/main/accuracy
     total [..................................................]  0.83%
this epoch [########..........................................] 16.67%
       100 iter, 0 epoch / 20 epochs
       inf iters/sec. Estimated time to finish: 0:00:00.
```

### CODE REVIEW

File: chainer/training/extensions/progress_bar.py [hunk 1]
```diff
@@ -26,6 +26,9 @@ class ProgressBar(extension.Extension):
         out: Stream to print the bar. Standard output is used by default.
 
     """
+
+    invoke_before_training = True
```
Comment (ir5) [inline chainer/training/extensions/progress_bar.py]: Well, why do we need to invoke this extension before training? When I ran mnist example, this progress bar was shown for just a moment at first, followed by `PrintReport` output. This was a little strange.
When I ran the example with `invoke_before_training = False`, the output was good. So I suspect that this is actually unnecessary.

**Comment by okuta** [comment]
Sorry, I changed this PR title.

File: chainer/training/extensions/progress_bar.py (hunk 1 above)
Comment (ir5) [inline chainer/training/extensions/progress_bar.py]: I'm not sure if you understand what I'm saying, so I put a screen capture here. You can see progress bar disappears after the `PrintReport` output.
aaa

Comment (okuta) [inline chainer/training/extensions/progress_bar.py]: Oh, I understood.
I will fix and change title.

**Comment by ir5** [comment]
Now the code looks good to me.

**Comment by beam2d** [comment]
LGTM

**Pull request merged**